invoicer_project/static/js/registration_request.js

```diff
@@ -0,0 +1,269 @@
+});
+
+emailField.addEventListener('input', () => {
+    emailField.removeAttribute("errorText");
+    emailField.removeAttribute("error");
+});
+
+async function actionAfterRegisterRequest(registerStatusCode, dataToSend) {
+    if (registerStatusCode === 200) {
+        const tokenObtainStatusCode = await obtainAndSaveTokens(host + '/user/login/', dataToSend);
+        if (tokenObtainStatusCode === 200) {
+            window.location.replace(host + '/clients/home/');
+        } else {
+            showBackEndErrorsAtFE(tokenObtainStatusCode);
+        }
+    } else {
+        emailField.setAttribute("error", "true");
+        passwordField.setAttribute("error", "true");
+        repeatPasswordField.setAttribute("error", "true");
+        if (registerStatusCode === 400) {
+            passwordField.setAttribute("errorText", "Invalid credentianls!");
+        } else if (registerStatusCode === 409) {
+            passwordField.removeAttribute("error");
+            repeatPasswordField.removeAttribute("error");
+            emailField.setAttribute("errorText", "User with such email already exists!");
+        } else {
+            passwordField.setAttribute("errorText", "Unknown server error");
+        }
+    }
+}
+
+function showBackEndErrorsAtFE(status) {
+    emailField.setAttribute("error", "true");
+    passwordField.setAttribute("error", "true");
+    if (status === 400) {
+        passwordField.setAttribute("errorText", "Invalid credentials");
+    } else {
+        passwordField.setAttribute("errorText", "Unknown error");
+    }
+}
+
+const obtainAndSaveTokens = async (url, dataToSend) => {
+    let statusCode, jsonResponse;
+    try {
+         const res = await fetch(url, {
+             method: 'POST',
+             body: dataToSend
+         })
+        jsonResponse = await res.json();
+        statusCode = res.status;
+    } catch (error) {
+        console.error(error);
+    } 
+    if (statusCode === 200) {
+            window.localStorage.setItem('accessToken', jsonResponse['access']);
+            window.localStorage.setItem('refreshToken', jsonResponse['refresh']);
+    }
+    return statusCode;
+}
+
+const registerNewUser = async (url, dataToSend) => {
+    let statusCode;
+    try {
+        const res = await fetch(url, {
+            method: 'POST',
+            body: dataToSend
+        });
+        statusCode = res.status;
+    } catch (error) {
+        console.error(error);
+    }
+    return statusCode;
+}
+
+document.getElementById("sign_up_confirm_btn_rg_pg").addEventListener("click", async function (event) {
+    event.preventDefault();
+    const validationFieldsList = validateRegistration();
+    if (allAreFalse(validationFieldsList)) {
+        const name = nameField.value;
+        const surname = surnameField.value;
+        const email = emailField.value;
+        const password = passwordField.value;
+        const repeat_password = repeatPasswordField.value;
+        const formData = new FormData();
```

Comment:
You could simplify this code by using simple javascript object instad of FormData

Reply:
unfortunately, django keeps returning 415 server error: unsupported media type
(it doesn't know how to parse stringified JS object shomehow)
TRIED:
1. Changing default django parsers
2. Modyfing headers parameter "Content-type" to 'application'json' 
3. Adding 'X-Requested-With': 'XMLHttpRequest' to request headers
Still getting 415 server response

Reply:
I have made [branch ](https://github.com/student-project-lnu-2022/invocer/pull/9)and have already send pull request with example how to work with json on client and server side

Reply:
done